File: src/main/java/me/sangbo/ps/Send.java
package me.sangbo.ps;

import com.rabbitmq.client.Channel;
import com.rabbitmq.client.Connection;
import me.sangbo.ConnectionUtils;

import java.io.IOException;
import java.util.concurrent.TimeoutException;

/**
 * Created by 桑博 on 2018/6/13.
 */
public class Send {

  public static final String EXCHANGE_NAME= "exchange_ps_sangbo_fanout";

  public static void main(String[] args) throws IOException, TimeoutException {


    Connection connection = ConnectionUtils.getConnection();
    Channel channel = connection.createChannel();
    channel.exchangeDeclare(EXCHANGE_NAME,"fanout");
    String msg = "分发";
    channel.basicPublish(EXCHANGE_NAME,"",null,msg.getBytes());

    System.out.println("send："+msg);
    channel.close();
    connection.close();

  }

}

File: src/main/java/me/sangbo/topic/Send.kt
package me.sangbo.topic

import me.sangbo.ConnectionUtils

/**
 * Created by 桑博 on 2018/6/14.
 */
private const val EXCHANGE_NAME = "exchange_sangbo_topic"
private const val ROULTING_KEY = "goods.delete";
fun main(args: Array<String>) {

    val connection = ConnectionUtils.getConnection()
    val channel = connection.createChannel()
    channel.exchangeDeclare(EXCHANGE_NAME,"topic");
    val msg = "kotlin";
    channel.basicPublish(EXCHANGE_NAME, ROULTING_KEY,null,msg.toByteArray())

    channel.close();
    connection.close();

}